libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.processor.ts

```diff
@@ -0,0 +1,175 @@
+    const documentQueryService = new DocumentQueryService(
+      provideDocumentLoaderService,
+    );
+
+    return documentQueryService.load({
+      context: {
+        s3KeyPrefix: ruleInput.documentKeyPrefix,
+      },
+      criteria: {
+        relatedDocuments: [this.certificateMatch.match],
+      },
+      documentId: ruleInput.documentId,
+    });
+  }
+
+  async getCreditsUnitPrice(ruleInput: RuleInput): Promise<NonZeroPositive> {
+    const creditsDocument = await loadParentDocument(
+      this.context.documentLoaderService,
+      toDocumentKey({
+        documentId: ruleInput.documentId,
+        documentKeyPrefix: ruleInput.documentKeyPrefix,
+      }),
+    );
+
+    if (isNil(creditsDocument)) {
+      throw this.errorProcessor.getKnownError(
+        this.errorProcessor.ERROR_MESSAGE.CREDITS_DOCUMENT_NOT_FOUND,
+      );
+    }
+
+    const unitPrice = getEventAttributeValue(
+      creditsDocument.externalEvents?.find((event) =>
+        eventHasName(event, RULES_METADATA),
+      ),
+      UNIT_PRICE,
+    );
+
+    if (!is<NonZeroPositive>(unitPrice) || new BigNumber(unitPrice).isNaN()) {
+      throw this.errorProcessor.getKnownError(
+        this.errorProcessor.ERROR_MESSAGE.INVALID_UNIT_PRICE,
+      );
+    }
+
+    return unitPrice;
+  }
+
+  getRewardsDistributionRuleValue(
+    massIdCertificateDocument: Document,
+  ): CertificateRewardDistributionOutput {
+    const rulesMetadataEvent = massIdCertificateDocument.externalEvents?.find(
+      eventNameIsAnyOf([RULES_METADATA]),
+    );
+
+    const rewardsDistributionRuleResultContent = getEventAttributeValue(
+      rulesMetadataEvent,
+      REWARDS_DISTRIBUTION_RULE_RESULT_CONTENT,
+    );
+
+    if (
+      !is<CertificateRewardDistributionOutput>(
+        rewardsDistributionRuleResultContent,
+      )
+    ) {
+      throw this.errorProcessor.getKnownError(
+        this.errorProcessor.ERROR_MESSAGE.REWARDS_DISTRIBUTION_RULE_RESULT_CONTENT_NOT_FOUND(
+          massIdCertificateDocument.id,
+        ),
+      );
+    }
+
+    return rewardsDistributionRuleResultContent;
+  }
+
+  async getRuleSubject(ruleInput: RuleInput): Promise<RuleSubject> {
+    const certificateDocumentsQuery =
+      await this.generateCertificateDocumentsQuery(ruleInput);
+
+    const resultContentsWithMassIdCertificateValue =
+      await certificateDocumentsQuery?.iterator().map(
+        ({ document }): ResultContentsWithMassIdCertificateValue => ({
+          massIdCertificateValue: new BigNumber(document.currentValue),
+          resultContent: this.getRewardsDistributionRuleValue(document),
+        }),
+      );
+
+    if (!isNonEmptyArray(resultContentsWithMassIdCertificateValue)) {
+      throw this.errorProcessor.getKnownError(
+        this.errorProcessor.ERROR_MESSAGE.CERTIFICATE_DOCUMENT_NOT_FOUND(
+          this.certificateMatch.match.type,
+        ),
+      );
+    }
+
+    return {
+      creditsDocumentUnitPrice: await this.getCreditsUnitPrice(ruleInput),
```

Comment:
I think a more suitable name here would be `creditUnitPrice`
```suggestion
      creditUnitPrice: await this.getCreditUnitPrice(ruleInput),
```